internal/wasi/wasi.go

```diff
@@ -504,6 +649,53 @@ func (a *wasiAPI) ClockTimeGet(ctx wasm.ModuleContext, id uint32, precision uint
 	return wasi.ErrnoSuccess
 }
 
+// FdClose implements SnaphotPreview1.FdClose
+func (a *wasiAPI) FdClose(ctx wasm.ModuleContext, fd uint32) wasi.Errno {
+	f, ok := a.opened[fd]
+	if !ok {
+		return wasi.ErrnoBadf
+	}
+
+	if f.file != nil {
+		f.file.Close()
+	}
+
+	delete(a.opened, fd)
+
+	return wasi.ErrnoSuccess
+}
+
+// FdPrestatGet implements SnahpshotPreview1.FdPrestatGet
+// TODO: Currently FdPrestatGet implements nothing except returning ErrnoBadf
+func (a *wasiAPI) FdPrestatGet(ctx wasm.ModuleContext, fd uint32, bufPtr uint32) wasi.Errno {
+	if _, ok := a.opened[fd]; !ok {
+		return wasi.ErrnoBadf
+	}
+	return wasi.ErrnoSuccess
+}
+
+// FdPrestatDirName implements SnahpshotPreview1.FdPrestatDirName
+func (a *wasiAPI) FdPrestatDirName(ctx wasm.ModuleContext, fd uint32, pathPtr uint32, pathLen uint32) wasi.Errno {
+	f, ok := a.opened[fd]
+	if !ok {
+		return wasi.ErrnoBadf
+	}
+
+	// The semantics of pathLen in wazero follows that of wasmtime.
+	// See https://github.com/bytecodealliance/wasmtime/blob/2ca01ae9478f199337cf743a6ab543e8c3f3b238/crates/wasi-common/src/snapshots/preview_1.rs#L578-L582
+	// Some runtimes such as wasmer may have another semantics, where they consider pathLen as the maximum length to write, instead of the exact length to write.
+	// See https://github.com/wasmerio/wasmer/blob/3463c51268ed551933392a4063bd4f8e7498b0f6/lib/wasi/src/syscalls/mod.rs#L764
```

Comment:
Added the comments

Reply:
put in RATIONALE.md as we are accumulating a lot of these and will accumulate more due to underspecification. Then, you can line comment or trailing comment the anchor text if you like.